kernel/src/main/scala/com/ocadotechnology/pass4s/kernel/Consumer.scala

```diff
@@ -206,8 +206,8 @@ object Consumer extends ConsumerInstances {
   /** [[cats.tagless.InvariantK]] instance for Consumer. The effect appears in both covariant and contravariant positions, so we can't get
     * anything stronger like [[cats.tagless.FunctorK]].
     */
-  implicit def invariantK[A]: InvariantK[Consumer[*[_], A]] =
-    new InvariantK[Consumer[*[_], A]] {
+  implicit def invariantK[A]: InvariantK[({ type C[F[_]] = Consumer[F, A] })#C] =
+    new InvariantK[({ type C[F[_]] = Consumer[F, A] })#C] {
```

Comment:
I think you can retain the old syntax by configuring compiler flags https://docs.scala-lang.org/scala3/guides/migration/plugin-kind-projector.html

Reply:
higher-kinded type lambda placeholders & higher-kinded named type lambda parameters are not supported as constructs - https://docs.scala-lang.org/scala3/guides/migration/plugin-kind-projector.html#rewriting-incompatible-constructs

I couldn't manage to make it work for `Consumer[*[_], A]`

Reply:
I was also eager to use the new Scala 3 syntax for type lambdas, but IMO - it's a bit too much to create separate source directories only for short syntax improvement. And it's only done in 2 places in the whole project.

Reply:
Sooner or later this needs to happen, but definitely out of scope of this PR. How about leaving a comment around this code, linking this discussion for future reference?

Reply:
Added in the last commit